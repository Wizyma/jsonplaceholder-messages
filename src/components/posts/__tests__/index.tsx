import { ReactQueryCacheProvider, makeQueryCache } from 'react-query'
import { render, cleanup, wait } from '@testing-library/react';
import React from 'react';

import Component from '..';
import PaginationProvider from '../../../context/pagination-context';
import postsMock from './mock.json';

beforeEach(() => {
  cleanup();
});

const queryCache = makeQueryCache()

function Providers({ children }: { children: any }) {

  return (
    <PaginationProvider>
      <ReactQueryCacheProvider queryCache={queryCache}>
        {children}
      </ReactQueryCacheProvider>
    </PaginationProvider>
  )
}

describe(`${Component.name} should pass all test`, () => {
  it('render posts with the loading state', () => {
    const { asFragment } = render(<Component />, {
      wrapper: ({ children }) => (
        <Providers>
          {children}
        </Providers>
      )
    })

    expect(asFragment()).toMatchSnapshot()
  })

  it('render posts with the query state', () => {
    const getPosts = () => postsMock
    queryCache.setQueryData(['posts', 1, 10], getPosts)
    const { asFragment } = render(<Component />, {
      wrapper: ({ children }) => (
        <Providers>
          {children}
        </Providers>
      )
    })
    
    wait(() => expect(asFragment()).toMatchSnapshot())
  })
})
