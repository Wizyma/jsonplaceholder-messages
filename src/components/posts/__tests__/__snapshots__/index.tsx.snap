// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts should pass all test render posts with the loading state 1`] = `
<DocumentFragment>
  <h2
    class="MuiTypography-root MuiTypography-h2"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
    />
  </h2>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
    />
  </p>
  <h2
    class="MuiTypography-root MuiTypography-h2"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
    />
  </h2>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
    />
  </p>
  <h2
    class="MuiTypography-root MuiTypography-h2"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
    />
  </h2>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
    />
  </p>
  <h2
    class="MuiTypography-root MuiTypography-h2"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
    />
  </h2>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
    />
  </p>
  <h2
    class="MuiTypography-root MuiTypography-h2"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
    />
  </h2>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
    />
  </p>
  <h2
    class="MuiTypography-root MuiTypography-h2"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
    />
  </h2>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
    />
  </p>
  <h2
    class="MuiTypography-root MuiTypography-h2"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
    />
  </h2>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
    />
  </p>
  <h2
    class="MuiTypography-root MuiTypography-h2"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
    />
  </h2>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
    />
  </p>
  <h2
    class="MuiTypography-root MuiTypography-h2"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
    />
  </h2>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
    />
  </p>
  <h2
    class="MuiTypography-root MuiTypography-h2"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
    />
  </h2>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
    />
  </p>
</DocumentFragment>
`;

exports[`Posts should pass all test render posts with the query state 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
        style="padding: 10px;"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5"
        >
           sunt aut facere repellat provident occaecati excepturi optio reprehenderit
        </h5>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto
        </p>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
        style="padding: 10px;"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5"
        >
           qui est esse
        </h5>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla
        </p>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
        style="padding: 10px;"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5"
        >
           ea molestias quasi exercitationem repellat qui ipsa sit aut
        </h5>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          et iusto sed quo iure
voluptatem occaecati omnis eligendi aut ad
voluptatem doloribus vel accusantium quis pariatur
molestiae porro eius odio et labore et velit aut
        </p>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
        style="padding: 10px;"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5"
        >
           eum et est occaecati
        </h5>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          ullam et saepe reiciendis voluptatem adipisci
sit amet autem assumenda provident rerum culpa
quis hic commodi nesciunt rem tenetur doloremque ipsam iure
quis sunt voluptatem rerum illo velit
        </p>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
        style="padding: 10px;"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5"
        >
           nesciunt quas odio
        </h5>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          repudiandae veniam quaerat sunt sed
alias aut fugiat sit autem sed est
voluptatem omnis possimus esse voluptatibus quis
est aut tenetur dolor neque
        </p>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
        style="padding: 10px;"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5"
        >
           dolorem eum magni eos aperiam quia
        </h5>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          ut aspernatur corporis harum nihil quis provident sequi
mollitia nobis aliquid molestiae
perspiciatis et ea nemo ab reprehenderit accusantium quas
voluptate dolores velit et doloremque molestiae
        </p>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
        style="padding: 10px;"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5"
        >
           magnam facilis autem
        </h5>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          dolore placeat quibusdam ea quo vitae
magni quis enim qui quis quo nemo aut saepe
quidem repellat excepturi ut quia
sunt ut sequi eos ea sed quas
        </p>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
        style="padding: 10px;"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5"
        >
           dolorem dolore est ipsam
        </h5>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          dignissimos aperiam dolorem qui eum
facilis quibusdam animi sint suscipit qui sint possimus cum
quaerat magni maiores excepturi
ipsam ut commodi dolor voluptatum modi aut vitae
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
